packages/jest-types/__typetests__/expect.test.ts

```diff
@@ -38,6 +38,21 @@ expectError(expect(0.1 + 0.2).toEqual(expect.not.closeTo('three')));
 expectError(expect(0.1 + 0.2).toEqual(expect.not.closeTo(0.3, false)));
 expectError(expect(0.1 + 0.2).toEqual(expect.not.closeTo()));
 
+expectType<void>(expect(20).toEqual(expect.numberGreaterThan(10)));
+expectError(expect(9).toEqual(expect.numberGreaterThan('10')));
+expectError(expect(9).toEqual(expect.numberGreaterThan(false)));
+
+expectType<void>(expect(10).toEqual(expect.numberGreaterThanOrEqualTo(10)));
+expectType<void>(expect(2).toEqual(expect.numberGreaterThanOrEqualTo(10)));
+expectError(expect(2).toEqual(expect.numberGreaterThanOrEqualTo('1')));
+
+expectType<void>(expect(5).toEqual(expect.numberLessThan(6)));
+expectError(expect(21).toEqual(expect.numberLessThan('20')));
+
+expectType<void>(expect(5).toEqual(expect.numberLessThanOrEqualTo(5)));
+expectType<void>(expect(3).toEqual(expect.numberLessThanOrEqualTo(5)));
+expectError(expect(11).toEqual(expect.numberLessThanOrEqualTo('10')));
```

Comment:
The type tests are not executed. The file is only type checked by passing it though TS compiler programatically. From that point of view some tests are identical and can be removed.

I have added tests for `.not` instead, but these are failing right now. You should make a change in typings of `@jest/expect` package similar to the one in `expect`.

```suggestion
expectType<void>(expect(20).toEqual(expect.numberGreaterThan(10)));
expectError(expect(9).toEqual(expect.numberGreaterThan(false)));
expectError(expect(20).toEqual(expect.not.numberGreaterThan(10)));

expectType<void>(expect(2).toEqual(expect.numberGreaterThanOrEqualTo(10)));
expectError(expect(2).toEqual(expect.numberGreaterThanOrEqualTo('1')));
expectError(expect(2).toEqual(expect.not.numberGreaterThanOrEqualTo(10)));

expectType<void>(expect(5).toEqual(expect.numberLessThan(6)));
expectError(expect(21).toEqual(expect.numberLessThan('20')));
expectError(expect(5).toEqual(expect.not.numberLessThan(6)));

expectType<void>(expect(5).toEqual(expect.numberLessThanOrEqualTo(5)));
expectError(expect(11).toEqual(expect.numberLessThanOrEqualTo('10')));
expectError(expect(5).toEqual(expect.not.numberLessThanOrEqualTo(5)));
```

Reply:
Thank you! I removed the duplicated tests, but I didn't add the tests for the inverse matchers (`expect.not`), because I had removed the new matchers from the `expect.not`. Should I add the new matchers to the `expect.not` then?

Reply:
In the example above I used `expectError` to test `expect.not` matchers. The idea is to show that these matchers do not exist on `expect` object. Indirect testing is not perfect, but that was the point. Currently these tests fail and that means that types of new matchers are still added to `expect.not`.

As I wrote, to fix this you should make a change in typings of `@jest/expect` package similar to the one in `expect`. (Keep them in `expect` as well, of course.)